### ISSUE TITLE
Invalid programs should show "0" (verification failed) instead of throwing an exception

### ISSUE THREAD

**Issue by dthaler** [vbpf/ebpf-verifier#129]
Sample program that should fail verification:
```
int func()
{
    while (1) {
    }
    return 0;
}
```

Currently this program will result in `instruction_seq_to_cfg()` throwing a C++ exception in asm_cfg.hpp line 45:
```
    if (exit_label.empty())
        throw std::runtime_error("no exit");
```

**Comment by elazarg** [comment]
Yep. We need a verification phase for program structure, as there used to be when it was a vbpf fork.

**Comment by elazarg** [comment]
For reference, ubpf validation is [here](https://github.com/iovisor/ubpf/blob/089f6279752adfb01386600d119913403ed326ee/vm/ubpf_vm.c#L566). I don't think it performs this particular check though.

**Comment by dthaler** [comment]
The case in https://github.com/vbpf/ebpf-verifier/pull/132 is another example of a C++ exception that should be converted (via try/catch) into a verification failure.

**Comment by elazarg** [comment]
Perhaps a different error code - I want this to be visible when running tests. I was using `1` for success and `0` for failure simply because it was easy to use with spreadsheets :)

**Comment by dthaler** [comment]
With PR #141 being merged (which removed the exception throw mentioned in this issue), this program now passes validation.

Hopefully once termination checking is done, it will gracefully fail.
